cube-builder-aws/cube_builder_aws/maestro.py

```diff
@@ -783,207 +826,210 @@ def publish(self, activity):
+                ).save()
+
+        # Register all ARD scenes - WARPED Collection
+        for datedataset in activity['scenes']:
+            scene = activity['scenes'][datedataset]
 
-        # delete 'assets' and 'collection_items' if exists
-        assets = Asset.query().filter(
-            Asset.collection_item_id == general_scene_id
-        ).all()
-        for asset in assets:
-            db.session().delete(asset)
-            db.session().commit()
-
-        coll_item = CollectionItem.query().filter(
-            CollectionItem.id == general_scene_id
-        ).first()
-        if coll_item:
-            db.session().delete(coll_item)
-            db.session().commit()
-
-        # insert 'collection_item'
-        pngname = '{}.png'.format(general_scene_id)
-        s3pngname = os.path.join(activity['dirname'], str(scene['date'])[0:10], pngname)
-        CollectionItem(
-            id=general_scene_id,
-            collection_id=cube_id,
-            grs_schema_id=cube.grs_schema_id,
-            tile_id=activity['tileid'],
-            item_date=scene['date'],
-            composite_start=scene['date'],
-            composite_end=scene['date'],
-            quicklook='{}/{}'.format(bucket_name, s3pngname),
-            cloud_cover=int(scene['cloudratio']),
-            scene_type='WARPED',
-            compressed_file=None
-        ).save()
-
-        # insert 'assets'
-        bands_by_cube = Band.query().filter(
-            Band.collection_id == cube_id
-        ).all()
-        for band in activity['bands']:
-            if band not in scene['ARDfiles']:
-                print('publish - problem - band {} not in scene[files]'.format(band))
-                continue
-            band_id = list(filter(lambda b: str(b.common_name) == band, bands_by_cube))
-            if not band_id:
-                print('band {} not found!'.format(band))
-                continue
-            
-            raster_size_x = scene['raster_size_x'] if scene.get('raster_size_x') else activity.get('raster_size_x')
-            raster_size_y = scene['raster_size_y'] if scene.get('raster_size_y') else activity.get('raster_size_y')
-            block_size = scene['block_size'] if scene.get('block_size') else activity.get('block_size')
-            Asset(
+            cube_id = get_cube_id(activity['datacube'])
+            cube = Collection.query().filter(
+                Collection.id == cube_id
+            ).first()
+            if not cube:
+                raise Exception('cube {} not found!'.format(cube_id))
+
+            general_scene_id = '{}_{}_{}'.format(
+                cube_id, activity['tileid'], str(scene['date'])[0:10])
+
+            # delete 'assets' and 'collection_items' if exists
+            assets = Asset.query().filter(
+                Asset.collection_item_id == general_scene_id
+            ).all()
+            for asset in assets:
+                db.session().delete(asset)
+                db.session().commit()
```

Comment:
Is really necessary to commit for each asset removal?

Maybe you can use a transaction save point on top of context like:
```python
with db.session.begin_nested():
    # delete item
    # delete X
    for asset in assets:
        db.session.delete(asset)
    
    # Add new assets, etc

db.session.commit()  # It not required to call rollback since the sqlalchemy already handle it on error using savepoint.
```

In this way, you can preserve the transaction id on database cluster.